### PR TITLE
#24644 : Clicking a Container as File is not taking you to the right location in the Site Browser

### LINKED ISSUE #24644: [Containers] : Clicking a Container as File is not taking you to the right location in the Site Browser
### Parent Issue

_No response_

### Problem Statement

In the `Containers` portlet, when users want to edit any Container as File, they click its name and are taken to the `Site Browser` portlet to the exact location of such a Container. This is not working right now.

### Steps to Reproduce

- Log into dotCMS, with the Full Starter.
- Go to the **Site > Containers** portlet.
- Click either the `default` or `banner` Container -- which are the only Containers as Files in the Full Starter.
- The **Site Browser** portlet is neither opening the expected folder nor displaying its contents.

### Acceptance Criteria

Clicking any Container as File must take you to the **Site Browser** portlet in its expected location.

### dotCMS Version

dotCMS 23.05

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Found when working on ticket: https://github.com/dotCMS/core/issues/23798

### Assumptions & Initiation Needs

There are two columns in the `folder` database table named `inode` and `identifier`. In recent versions, when you create a folder, both columns have the value of the folder's Identifier.

However, in older versions, a folder had a specific Inode and Identifier, which were not the same. We need to fix this inconsistency. Maybe an Upgrade Task can take care of this.

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### <samp>🤖 Generated by Copilot at 954cbd6</samp>

The pull request improves the functionality and code quality of the browser tree feature, which allows users to navigate and select folders and sites in the dotCMS UI. It refactors the `selectFolder` method in `BrowserTreeHelper.java` and the `BrowserAjax.java` class, and adds new utility methods to `DwrUtil.java` for folder tree operations. It also applies code style and documentation standards to the modified files.

### <samp>🤖 Generated by Copilot at 954cbd6</samp>

*  Reorder and add imports to follow code style guidelines and use new classes ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-ed592e4d521552ac42576b554030295927c520d45a10994232e9fdf48f408f5eL3-R8), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-ed592e4d521552ac42576b554030295927c520d45a10994232e9fdf48f408f5eL18), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-ed592e4d521552ac42576b554030295927c520d45a10994232e9fdf48f408f5eL24-R23), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-e3b7aa4830b195711eeebbf00eebe65af56103156b2fcc08aed92a82546895cdL6-R23), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L22-R24))
*  Add constants for session attribute names related to folder tree state ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-ed592e4d521552ac42576b554030295927c520d45a10994232e9fdf48f408f5eR38-R40), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9R85-R86))
*  Add utility methods to DwrUtil.java to get user roles, permissions, and session from web context ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-e3b7aa4830b195711eeebbf00eebe65af56103156b2fcc08aed92a82546895cdL97-R157))
*  Modify selectFolder method in BrowserTreeHelper.java to use new constants, utility methods, and logic to handle folder and site selection ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-ed592e4d521552ac42576b554030295927c520d45a10994232e9fdf48f408f5eL143-R172))
*  Modify getTree method in BrowserAjax.java to use new constants, utility methods, and logic to get the folder tree for a host or folder ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L144-R192))
*  Remove unused setCurrentOpenFolder method in BrowserAjax.java ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L279-L307))
*  Modify getFolders method in BrowserAjax.java to use new constants, utility methods, and logic to get the contents of a site or folder ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L339-R300), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L422-R378), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L437-R391))
*  Modify getFoldersTree methods in BrowserAjax.java to use new utility methods, logic, and javadoc comments to get the sub-folders of a host or folder ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L849-R832), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L873-R881))
*  Remove private getUserRoles method in BrowserAjax.java as it is replaced by the utility method in DwrUtil.java ([link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L2311-R2285), [link](https://github.com/dotCMS/core/pull/24664/files?diff=unified&w=0#diff-0947ca188a7d8785100a58970fa5567353acd19519d555eb51cdf70a257864f9L2525-L2542))

### CODE REVIEW

**Comment by jdotcms** [comment]
The code looks good, but would need some test

**Review by jdotcms — APPROVED** [review]
Good job

**Review by jgambarios — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/browsertree/BrowserTreeHelper.java [hunk 1]
```diff
@@ -87,15 +88,13 @@ public List<Treeable> getTreeablesUnder(String siteName, User user, String uri)
     /**
      * Tries to find the select folder
      * @return Optional String, present if the folder select exists
-     * @throws DotDataException
-     * @throws DotSecurityException
      */
-    public Optional<String> findSelectedFolder (final HttpServletRequest request) throws DotDataException, DotSecurityException {
+    public Optional<String> findSelectedFolder (final HttpServletRequest request) {
 
         final BrowserAjax browserAjax = (BrowserAjax)request.getSession().getAttribute("BrowserAjax");
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/browsertree/BrowserTreeHelper.java:94]: We should not be storing the "BrowserAjax" class in session at all.  We should only store the active / open `folderId`

**Review by wezell — CHANGES_REQUESTED** [review]
Hard to review due to formatting changes.  Are we sure this gets the BrowserAjax.class out of our session?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/browsertree/BrowserTreeHelper.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/browsertree/BrowserTreeHelper.java]: Done! As we're using the Session now, this method is not necessary anymore.

**Comment by jcastro-dotcms** [comment]
> Hard to review due to formatting changes. Are we sure this gets the BrowserAjax.class out of our session?

@wezell I looked for references of the Session object getting an instance of the `BrowserAjax` class and the actual "BrowserAjax" String in our code base, and couldn't find any.

**Review by wezell — APPROVED** [review]

**Pull request merged**